curation-migration-datasync/eventConsumer.ts

```diff
@@ -28,3 +28,32 @@ export async function addScheduledItem(
   const curatedItemModel = new CuratedItemRecordModel();
   await curatedItemModel.insertFromEvent(curatedRecId, eventBody);
 }
+
+/**
+ * Remove scheduled item from dynamoDB mapping and legacy database
+ * @param eventBody event payload; only need scheduledItemExternalId
+ * @param db db connection
+ */
+export async function removeScheduledItem(
+  eventBody: ScheduledItemPayload,
+  db: Knex
+) {
+  const dbService = new DataService(db);
+  const curatedItemModel = new CuratedItemRecordModel();
+  const curatedRecord = await curatedItemModel.getByScheduledItemExternalId(
+    eventBody.scheduledItemExternalId
+  );
+  if (curatedRecord.length === 0) {
+    throw new Error(
+      `No mapping found for scheduledItemExternalId=${eventBody.scheduledItemExternalId}`
+    );
+  }
+  // TODO: Confused about why result of getByScheduledItemExternalId is an array; isn't it only 1:1?
```

Comment:
the dynamo query command with global secondary index always returns an arary instead of one value.
https://github.com/Pocket/curation-tools-data-sync/blob/bfdbef3b6a0dcfd2c8e3c9c908d66b4c18d65f74/curation-migration-backfill/dynamodb/curatedItemIdMapper.ts#L171

 yes, for our use case - we have 1:1 mapping of scheduledItem's `externalId` with curated_rec_id, we can modify the `getByScheduledItemExternalId` to return `res.Items.first()`